### PR TITLE
Fixed #34983 -- Deprecated django.utils.itercompat.is_iterable().

### PR DESCRIPTION
**Pull request by ngnpope**
See ticket-34983.

### CODE REVIEW

**Comment by ngnpope** [comment]
I did a search for try-iter-except cases not using `is_iterable()` and found the following:

```
django/core/serializers/base.py:347:        pks_iter = iter(field_value)
django/http/multipartparser.py:675:        iterator = iter(stream_or_iterable)
django/http/response.py:485:            self._iterator = iter(value)
django/http/response.py:495:            return iter(self.streaming_content)
django/template/defaultfilters.py:691:                        iter(next_item)
django/urls/resolvers.py:737:            iter(patterns)
django/utils/http.py:62:                itr = iter(value)
tests/prefetch_related/tests.py:447:            iter(obj_iter)
```

I'm not sure whether we want to switch _some_ of these to use `isinstance(..., Iterable)` also?

File: docs/releases/5.1.txt [hunk 1]
```diff
@@ -292,6 +292,11 @@ Miscellaneous
   ``ModelAdmin.log_deletions()`` and  ``LogEntryManager.log_actions()``
   instead.
 
+* The ``django.utils.itercompat.is_iterable()`` function is deprecated. Use
+  ``isinstance(..., collections.abc.Iterable)`` instead. The module
+  ``django.utils.itercompat`` module will also be removed as this is the last
+  function remaining.
```
Comment (felixxm) [inline docs/releases/5.1.txt:298]:
```suggestion
* The undocumented ``django.utils.itercompat.is_iterable()`` function and the
  ``django.utils.itercompat`` module are deprecated. Use
  ``isinstance(..., collections.abc.Iterable)`` instead.
```

**Review by felixxm — COMMENTED** [review]
@ngnpope Thanks :+1: I left small comments.

**Comment by felixxm** [comment]
> I did a search for try-iter-except cases not using `is_iterable()` and found the following:
> 
> ```
> django/core/serializers/base.py:347:        pks_iter = iter(field_value)
> django/http/multipartparser.py:675:        iterator = iter(stream_or_iterable)
> django/http/response.py:485:            self._iterator = iter(value)
> django/http/response.py:495:            return iter(self.streaming_content)
> django/template/defaultfilters.py:691:                        iter(next_item)
> django/urls/resolvers.py:737:            iter(patterns)
> django/utils/http.py:62:                itr = iter(value)
> tests/prefetch_related/tests.py:447:            iter(obj_iter)
> ```
> 
> I'm not sure whether we want to switch _some_ of these to use `isinstance(..., Iterable)` also?

Don't know if it's worth doing :thinking: There are also some `hasattr(..., "__iter__")` checks. We can discuss this separately.

**Comment by ngnpope** [comment]
> We can discuss this separately.

That's fair. Will open a follow up PR.

**Comment by felixxm** [comment]
@ngnpope Thanks :+1:

**Comment by ngnpope** [comment]
> > We can discuss this separately.
> 
> That's fair. Will open a follow up PR.

See #17523.

**Pull request merged**